DSESwiftTests/DSESwiftTests.swift

```diff
@@ -0,0 +1,412 @@
+            
+            sensor = Sensor(name: "sensor2", sensorOptions: sensorOptions, userId: "user1", source: sourceName, dataType: "JSON", synced: false)
+            try DatabaseHandler.insertSensor(sensor)
+            
+            sensors = DatabaseHandler.getSensors(sourceName)
+            XCTAssertEqual(sensors.count, 2)
+        }catch{
+            XCTFail("Exception was captured. Abort the test.")
+        }
+    }
+    
+    func testInsertSameSensorTwice() {
+        let sensorOptions = SensorOptions(meta: "", uploadEnabled: true, downloadEnabled: true, persist: true)
+        do{
+            let sourceName = "testSource"
+            let sensor = Sensor(name: "sensor1", sensorOptions: sensorOptions, userId: KeychainWrapper.stringForKey(KEYCHAIN_USERID)!, source: sourceName, dataType: "JSON", synced: false)
+            try DatabaseHandler.insertSensor(sensor)
+            try DatabaseHandler.insertSensor(sensor)
+            
+        }catch{
+            print((error))
+            XCTAssertNotNil(error)
+        }
+    }
+    
+    func testInsertSensorWithInvalidUserId() {
+        let sensorOptions = SensorOptions(meta: "", uploadEnabled: true, downloadEnabled: true, persist: true)
+        do{
+            let sensor = Sensor(name: "sensor1", sensorOptions: sensorOptions, userId: "anotheruser", source: "testSource", dataType: "JSON", synced: false)
+            try DatabaseHandler.insertSensor(sensor)
+        }catch{
+            XCTAssertNotNil(error)
+        }
+    }
+    
+    func testUpdateSensor() {
+        let sensorOptions = SensorOptions(meta: "", uploadEnabled: true, downloadEnabled: true, persist: true)
+        do{
+            let sourceName = "testSource"
+            let sensor = Sensor(name: "sensor1", sensorOptions: sensorOptions, userId: KeychainWrapper.stringForKey(KEYCHAIN_USERID)!, source: "testSource", dataType: "JSON", synced: false)
+            try DatabaseHandler.insertSensor(sensor)
+            
+            let newSensorName = "sensor1Updated"
+            sensor.name = newSensorName
+            try DatabaseHandler.update(sensor)
+            
+            let retrievedSensor = try! DatabaseHandler.getSensor(sourceName, newSensorName)
+            XCTAssertEqual(retrievedSensor.name, newSensorName)
+        }catch{
+            XCTFail("Exception was captured. Abort the test.")
+        }
+    }
+    
+    func testUpdateSensorWithInvalidUsername() {
+        let sensorOptions = SensorOptions(meta: "", uploadEnabled: true, downloadEnabled: true, persist: true)
+        do{
+            let sensor = Sensor(name: "sensor1", sensorOptions: sensorOptions, userId: KeychainWrapper.stringForKey(KEYCHAIN_USERID)!, source: "testSource", dataType: "JSON", synced: false)
+            try DatabaseHandler.insertSensor(sensor)
+            
+            let newSensorName = "sensor1Updated"
+            sensor.name = newSensorName
+            sensor.userId = "AnotherUser"
```

Comment:
This is an invalid test case. update method will throw an exception if you give a different userId from the current user.